Sources/Operators/RemoveKnownDuplicates.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  Created by Jasdev Singh on 21/03/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+public extension Publisher where Output: Hashable {
+    /// When `Output` conforms to `Hashable`, `Publisher.removeKnownDuplicates` de-duplicates _all_
+    /// published value events, as opposed to pairwise with `Publisher.removeDuplicates`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Set`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events.
+    ///
+    /// - returns: A type-erased publisher that consumes duplicate values across all previous emissions from upstream.
+    func removeKnownDuplicates() -> AnyPublisher<Output, Failure> {
+        scan((Set<Output>(), Output?.none)) { seenAndPrevious, incoming in
+            var seen = seenAndPrevious.0
+            let isUnique = seen.insert(incoming).inserted
+            return (seen, isUnique ? incoming : nil)
+        }
+        .compactMap { $0.1 }
+        .eraseToAnyPublisher()
+    }
+}
+
+public extension Publisher where Output: Equatable {
+    /// When `Output` conforms to `Equatable`, `Publisher.removeKnownDuplicates` de-duplicates _all_
+    /// published value events, as opposed to pairwise with `Publisher.removeDuplicates`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Array`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events.
+    ///
+    /// - returns: A type-erased publisher that consumes duplicate values across all previous emissions from upstream.
+    func removeKnownDuplicates() -> AnyPublisher<Output, Failure> {
+        scan(([Output](), Output?.none)) { seenAndPrevious, incoming in
+            var seen = seenAndPrevious.0
+
+            guard !seen.contains(incoming) else { return (seen, nil) }
+
+            seen.append(incoming)
+            return (seen, incoming)
+        }
+        .compactMap { $0.1 }
+        .eraseToAnyPublisher()
+    }
+}
+
+public extension Publisher {
+    /// `Publisher.removeKnownDuplicates(by:)` de-duplicates _all_ published value events,
+    /// along the provided `by` predicate, as opposed to pairwise with `Publisher.removeDuplicates(by:)`.
+    ///
+    /// - note: It’s important to note that this operator stores all emitted values
+    ///         in an in-memory `Array`. So, use this operator with caution, when handling publishers
+    ///         that emit a large number of unique value events (as per `by`).
+    ///
+    /// - returns: A type-erased publisher that consumes duplicate values across all previous emissions from upstream
+    ///            (signaled with `by`).
+    func removeKnownDuplicates(by predicate: @escaping (Output) -> Bool) -> AnyPublisher<Output, Failure> {
```

Comment:
```suggestion
    func removeKnownDuplicates(by predicate: @escaping (Output) -> Bool) -> AnyPublisher<Output, Failure> {
```

I'm wondering here, is it not possible to implement the Equatable and Hashable ones by simply using this one and implementing the predicate? I know it's not like that in Ext but wondering if it's possible to save some _duplicate_ [pun-intended] implementation 🤔

Reply:
oo great call, let me give it a try!

Reply:
It worked! A note though, I had to refactor the `by` predicate to follow [Combine’s lead](https://developer.apple.com/documentation/combine/publisher/3204746-removeduplicates) since the only way to lean on `==` is to make the predicate two-ary.

Reply:
That's ok :)